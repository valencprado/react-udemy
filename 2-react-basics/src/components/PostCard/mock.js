export const postCardMock = {
    title: 'lorem ipsum dolor',
    body: 'body 1',
    id: 1,
    cover: 'img/img.png'
}
