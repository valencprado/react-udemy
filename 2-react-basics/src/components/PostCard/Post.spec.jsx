/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import { PostCard } from "."
import { render, screen } from "@testing-library/react";
import { postCardMock } from "./mock";

const props = postCardMock;

describe("<PostCard />", () => {
    it('should render PostCard correctly', () => {
    render(<PostCard {...props}/>);
    expect(screen.getByRole('img', {name: props.title}))
    .toHaveAttribute('src', 'img/img.png');
    expect(screen.getByRole('heading', /title 1/i)).toBeInTheDocument();
    expect(screen.getByText(/body/i)).toBeInTheDocument();
});
it('should match snapshot', () => {
        const {container} = render(<PostCard {...props}/>);
        expect(container.firstChild).toMatchSnapshot();
    })
})
