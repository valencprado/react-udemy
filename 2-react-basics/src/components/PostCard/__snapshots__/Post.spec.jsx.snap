// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="lorem ipsum dolor"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      lorem ipsum dolor
    </h2>
    <p>
      body 1
    </p>
  </div>
</div>
`;
